Sources/HummingbirdLambda/APIGatewayLambda.swift

```diff
@@ -20,18 +20,27 @@ import NIOHTTP1
 
 extension HBLambda where Event == APIGatewayRequest {
     /// Specialization of HBLambda.request where `In` is `APIGateway.Request`
-    public func request(context: LambdaContext, application: HBApplication, from: Event) throws -> HBRequest {
-        var request = try HBRequest(context: context, application: application, from: from)
-        // store api gateway request so it is available in routes
-        request.extensions.set(\.apiGatewayRequest, value: from)
-        return request
+    public func request(context: LambdaContext, from: Event) throws -> HBRequest {
+        return try HBRequest(context: context, from: from)
     }
 }
 
 extension HBLambda where Output == APIGatewayResponse {
     /// Specialization of HBLambda.request where `Out` is `APIGateway.Response`
-    public func output(from response: HBResponse) -> Output {
-        return response.apiResponse()
+    public func output(from response: HBResponse) async throws -> Output {
+        return try await response.apiResponse()
+    }
+}
+
+extension HBLambda where Event == APIGatewayRequest, Output == APIGatewayResponse, Context == APIGatewayRequestContext {
+    public func requestContext(
+        coreContext: HBCoreRequestContext,
```

Comment:
Instead of constructing the context like this. Why don't you create a new protocol for `HBLambdaRequestContext` which all lambda request contexts are required to conform to.
```swift
protocol HBLambdaRequestContext {
    associatedtype Event
    init<Event>(_ event: Event, applicationContext: HBApplicationContext, logger: Logger)
}
```
Then instead of calling `HBLambda.requestContext(coreContext:context:from:)` you call `HBLambda.Context(_:coreContext:)`.

This will make it easier for people to provide their own context types.

Reply:
Good point